pyne/endl.py

```diff
@@ -0,0 +1,239 @@
+                e_int = np.insert(e_int[e_int > low], 0, low)
+            if high in e_int:
+                xs = xs[e_int <= high]
+                e_int = e_int[e_int <= high]
+            elif high is not None:
+                high_xs = interp(np.log(high))
+                xs = np.append(xs[e_int < high], high_xs)
+                e_int = np.append(e_int[e_int < high], high)
+
+        de_int = float(e_int[-1]-e_int[0])
+        x1 = e_int[:-1]
+        x2 = e_int[1:]
+        y1 = xs[:-1]
+        y2 = xs[1:]
+        A = (y1-y2)/(np.log(x1/x2))
+        B = y1-A*np.log(x1)
+        return np.nansum(A*(x2*np.log(x2) -
+                            x1*np.log(x1)-x2+x1) + B*(x2-x1))/de_int
+
+    def _loglin(self, e_int, xs, low, high):
+        if low is not None or high is not None:
+            interp = interp1d(e_int, np.log(xs))
+            if low in e_int:
+                xs = xs[e_int >= low]
+                e_int = e_int[e_int >= low]
+            elif low is not None and low > e_int[0]:
+                low_xs = np.e ** interp(low)
+                xs = np.insert(xs[e_int > low], 0, low_xs)
+                e_int = np.insert(e_int[e_int > low], 0, low)
+            if high in e_int:
+                xs = xs[e_int <= high]
+                e_int = e_int[e_int <= high]
+            elif high is not None:
+                high_xs = np.e ** interp(high)
+                xs = np.append(xs[e_int < high], high_xs)
+                e_int = np.append(e_int[e_int < high], high)
+
+        de_int = float(e_int[-1]-e_int[0])
+        x1 = e_int[:-1]
+        x2 = e_int[1:]
+        y1 = xs[:-1]
+        y2 = xs[1:]
+        A = (np.log(y1)-np.log(y2))/(x1-x2)
+        B = np.log(y1) - A*x1
+        return np.nansum((y2-y1)/A)/de_int
+
+    def _loglog(self, e_int, xs, low, high):
+        if low is not None or high is not None:
+            interp = interp1d(np.log(e_int), np.log(xs))
+            if low in e_int:
+                xs = xs[e_int >= low]
+                e_int = e_int[e_int >= low]
+            elif low is not None and low > e_int[0]:
+                low_xs = np.e ** interp(np.log(low))
+                xs = np.insert(xs[e_int > low], 0, low_xs)
+                e_int = np.insert(e_int[e_int > low], 0, low)
+            if high in e_int:
+                xs = xs[e_int <= high]
+                e_int = e_int[e_int <= high]
+            elif high is not None:
+                high_xs = np.e ** interp(np.log(high))
+                xs = np.append(xs[e_int < high], high_xs)
+                e_int = np.append(e_int[e_int < high], high)
+
+        de_int = float(e_int[-1]-e_int[0])
+        x1 = e_int[:-1]
+        x2 = e_int[1:]
+        y1 = xs[:-1]
+        y2 = xs[1:]
+        A = - np.log(y2/y1)/np.log(x1/x2)
+        B = - (np.log(y1)*np.log(x2) - np.log(y2)*np.log(x1))/np.log(x1/x2)
+        return np.nansum(np.e**B / (A+1) * (x2**(A+1) - x1**(A+1))/de_int)
+
+
+
+class NotFound(Exception):
```

Comment:
This class isn't used anywhere and probably should be removed.  Also, I am wary of new exceptions :).